src/enzo/ExternalBoundary_SetGalaxySimulationBoundary.C

```diff
@@ -269,8 +322,26 @@ int ExternalBoundary::SetGalaxySimulationBoundary(FLOAT time)
 		}
 
 		// update metallicity field
+                if (MultiSpecies > 0)  {
+                  BoundaryValue[HIINum][dim][0][index] = 0.5 * CoolData.HydrogenFractionByMass * BoundaryValue[DensNum][dim][0][index];
+                  BoundaryValue[HeIINum][dim][0][index] = 0.2 * BoundaryValue[DensNum][dim][0][index] * (1.0 - CoolData.HydrogenFractionByMass); //get rid of factor of 4
+                  BoundaryValue[HeIIINum][dim][0][index] = 0.2 * BoundaryValue[DensNum][dim][0][index] * (1.0 - CoolData.HydrogenFractionByMass); //get rid of factor of 4
+                  BoundaryValue[HeINum][dim][0][index] = (1.0-CoolData.HydrogenFractionByMass)*BoundaryValue[DensNum][dim][0][index] - BoundaryValue[HeIINum][dim][0][index] - BoundaryValue[HeIIINum][dim][0][index];
+                  BoundaryValue[HINum][dim][0][index] = CoolData.HydrogenFractionByMass * BoundaryValue[DensNum][dim][0][index] - BoundaryValue[HIINum][dim][0][index];
+                  BoundaryValue[DeNum][dim][0][index] = BoundaryValue[HIINum][dim][0][index] + 0.25*BoundaryValue[HeIINum][dim][0][index] + 0.5*BoundaryValue[HeIIINum][dim][0][index];
+                  if (MultiSpecies > 1) {
+                    BoundaryValue[HMNum][dim][0][index] =1e-10;
```

Comment:
Is it possible to not fix these values in order to prevent unexpected behavior later? Or is this the standard "initial value" for these species.

Reply:
We have to set the boundary values to something (otherwise they could be undefined and corrupt the solution). Properly, these should be controlled with a parameter, but right now there is no way to know their true value. This code sets them to something vaguely appropriate -- generally, the wind flowing in is hot, so the ionization fractions will rapidly adjust to their correct equilibrium value, so in practice, the exact values here are not essential (but they need to be set). I will add a comment nothing this.

Reply:
I think leaving a comment explaining that the values should rapidly equilibrate is a good solution.